### PR TITLE
resolves #2060 set match on skip and scan_until

### LINKED ISSUE #2060: StringScanner returns nil on most recent match
Consider the following code:

```rb
require 'strscan'

stream = StringScanner.new('cat builder.rb')
if stream.skip(/cat/)
  puts stream[0].inspect
end
```
**MRI 2.6.3**

```
$ ruby test.rb  
"cat"
```

**Opal master (01f63335)**

```
$ bundle exec opal test.rb  
nil
```

### PR DESCRIPTION
**Pull request by Mogztter**
resolves #2060

### CODE REVIEW

File: stdlib/strscan.rb [hunk 1]
```diff
@@ -80,6 +82,16 @@ def scan_until(pattern)
   end
 
   def [](idx)
+    if @match.empty?
```
Comment (Mogztter) [inline stdlib/strscan.rb]: I took inspiration from MRI: https://github.com/ruby/ruby/blob/4c5eac73238a203830d43bccf22eee27d242616f/ext/strscan/strscan.c#L1165

**Comment by Mogztter** [comment]
@elia Do you want me to fix the codeclimate issues?

**Comment by Mogztter** [comment]
To be honest, I don't really see the code duplication... is that a bug?

File: stdlib/strscan.rb [hunk 2]
```diff
@@ -80,6 +82,16 @@ def scan_until(pattern)
...
+      return nil
+    end
+    if idx === Symbol
```
Comment (elia) [inline stdlib/strscan.rb]: Why not a case/when here?

```rb
case idx
when Symbol
…
```

Comment (Mogztter) [reply stdlib/strscan.rb]: Done :+1:

**Comment by Mogztter** [comment]
> Also can you change the commit message subject to just describe the change?
You can keep the "Resolves" in the commit description if you want +1

Sorry I'm so used to the Asciidoctor Git commit convention :sweat_smile: 
I've edited the commit subject and I've added an item in the CHANGELOG.

**Pull request merged**